components/header/podIndicator/index.jsx

```diff
@@ -19,55 +19,97 @@
+  event,
+  url,
+  router
+) => {
+  event.preventDefault();
+  const containerUrl = normalizeContainerUrl(url);
+  await router.replace("/resource/[iri]", resourceHref(containerUrl));
+  handleClose();
+  setUrl("");
+};
+
 export default function PodIndicator() {
   const classes = useStyles();
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const [url, setUrl] = useState("");
   const router = useRouter();
   const bem = useBem(useStyles());
-  const decodedResourceUri = decodeURIComponent(router.query.iri);
-  const [podUri, setPodUri] = useState();
-
-  useEffect(() => {
-    if (decodedResourceUri === "undefined") {
-      return;
-    }
-    const originUri = decodedResourceUri && new URL(decodedResourceUri).origin;
-    const decodedPodUri = decodeURIComponent(originUri);
-    setPodUri(decodedPodUri);
-  }, [decodedResourceUri]);
-
-  const { profile: podOwner } = usePodOwnerProfile(podUri);
-  if (!podOwner) {
-    return null;
-  }
+  const { profile, error } = usePodOwnerProfile();
+  const loading = !profile && !error;
 
-  const podOwnerName =
-    getStringNoLocale(podOwner.dataset, vcard.fn) ||
-    getStringNoLocale(podOwner.dataset, foaf.name);
+  const open = Boolean(anchorEl);
+  const id = open ? "pod-navigator" : undefined;
+  const handleClick = clickHandler(setAnchorEl);
+  const handleClose = closeHandler(setAnchorEl);
+  const onSubmit = submitHandler(handleClose, setUrl);
 
   return (
     <div className={classes.indicator}>
       <label htmlFor="pod-indicator-prompt" className={classes.indicatorLabel}>
         <span>Pod: </span>
-        <button
-          data-testid="pod-indicator-prompt"
-          type="button"
-          className={clsx(bem("button", "prompt"), classes.indicatorPrompt)}
+        {loading ? (
+          <Skeleton width={100} style={{ display: "inline-block" }} />
+        ) : (
+          <button
+            data-testid="pod-indicator-prompt"
+            id="pod-indicator-prompt"
+            type="button"
+            aria-describedby={id}
+            onClick={handleClick}
+            className={clsx(bem("button", "prompt"), classes.indicatorPrompt)}
+          >
+            {profile ? profile.name : "Unknown"}
+          </button>
+        )}
+        <Popover
+          id={id}
+          classes={{
+            paper: classes.popover,
+          }}
+          open={open}
+          anchorEl={anchorEl}
+          onClose={handleClose}
+          anchorOrigin={{
+            vertical: "bottom",
+            horizontal: "left",
+          }}
+          transformOrigin={{
+            vertical: "top",
+            horizontal: "left",
+          }}
         >
-          {podOwnerName}
-        </button>
+          <Form onSubmit={(event) => onSubmit(event, url, router)}>
+            <Input
+              id="PodNavigator"
+              label="Go to Pod"
+              value={url}
+              onChange={(event) => setUrl(event.target.value)}
```

Comment:
Add `type="url"` to get validation for free.

Reply:
Didn't use required (it sets the input to `:invalid` from the start), but checks it as part of submitting form.